crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -0,0 +1,129 @@
+use flox_types::catalog::CatalogEntry;
+use runix::flake_ref::git::GitRef;
+use runix::flake_ref::{protocol, FlakeRef};
+use runix::installable::AttrPath;
+use thiserror::Error;
+
+use crate::flox::Flox;
+use crate::providers::git::{GitCommandProvider as Git, GitProvider};
+
+/// State for the publish algorihm
+#[allow(dead_code)] // until we implement methods for Publish
+pub struct Publish<'flox> {
+    flox: &'flox Flox,
+    /// The published _upstream_ source
+    publish_ref: PublishRef,
+    /// The published attrpath
+    /// Should be fully resolved to avoid ambiguity
+    attr_path: AttrPath,
+    analysis: Option<CatalogEntry>, // model as type state?
+}
+
+impl<'flox> Publish<'flox> {
+    pub async fn new(
+        flox: &'flox Flox,
+        publish_ref: PublishRef,
+        attr_path: AttrPath,
+    ) -> PublishResult<Publish<'flox>> {
+        Ok(Self {
+            flox,
+            publish_ref,
+            attr_path,
```

Comment:
The bash impl seems to translate hydraJobs outputs into packages attrPaths?

https://github.com/flox/flox/blob/e5793cfddae8cfb6db3429696f2ac63ac4586728/flox-bash/lib/commands/publish.sh#L338

Though it also questions whether thats needed..
Lets skip that for  now